node/src/components/consensus/era_supervisor.rs

```diff
@@ -457,58 +467,30 @@ where
 
         for era_id in self.iter_past(self.current_era, self.bonded_eras().saturating_mul(2)) {
-            let new_faulty;
-            let validators;
-            let start_height;
-            let era_start_time;
-            let seed;
-
             let booking_block_hash = booking_blocks
                 .get(&era_id)
                 .expect("should have booking block");
 
+            let key_block = key_blocks.get(&era_id).expect("missing key block");
             #[allow(clippy::integer_arithmetic)] // Block height should never reach u64::MAX.
-            if era_id.is_genesis() {
-                new_faulty = vec![];
-                // The validator set was read from the global state: there's no key block for era 0.
-                validators = activation_era_validators.clone();
-                start_height = 0;
-                era_start_time = self
-                    .protocol_config
-                    .genesis_timestamp
-                    .expect("must have genesis start time if era ID is 0");
-                seed = 0;
-            } else {
-                // If this is not era 0, there must be a key block for it.
-                let key_block = key_blocks.get(&era_id).expect("missing key block");
-                start_height = key_block.height() + 1;
-                era_start_time = key_block.timestamp();
-                seed = Self::era_seed(*booking_block_hash, key_block.accumulated_seed());
-                if era_id == self.protocol_config.last_activation_point {
-                    // After an upgrade or emergency restart, we don't track faults cross-era.
-                    new_faulty = vec![];
-                    // And we read the validator sets from the global state, because the key block
-                    // might have been overwritten by the upgrade/restart.
-                    validators = activation_era_validators.clone();
-                } else {
-                    // If it's neither genesis nor upgrade nor restart, we use the validators from
-                    // the key block and ban validators that were faulty in previous eras.
-                    new_faulty = key_block
-                        .era_end()
-                        .expect("key block must be a switch block")
-                        .equivocators
-                        .clone();
-                    validators = key_block
-                        .next_era_validator_weights()
-                        .expect("missing validators from key block")
-                        .clone();
-                }
-            }
+            let start_height = key_block.height() + 1;
+            let era_start_time = key_block.timestamp();
+            let seed = Self::era_seed(*booking_block_hash, key_block.accumulated_seed());
+            // If it's neither genesis nor upgrade nor restart, we use the validators from
+            // the key block and ban validators that were faulty in previous eras.
```

Comment:
This comment doesn't make as much sense as it did as from the perspective of `EraSupervisor` it's never a _genesis or upgrade or restart_ since we're not creating an additional switch block when it _is_ one of these situations.

Reply:
Removed in 69fcec87f